### PR TITLE
Update `github.com/smallstep/nosql` to v0.5.0

### PR DESCRIPTION
**Pull request by hslatman**
This version of `nosql` includes some fixes for MySQL DSNs: https://github.com/smallstep/nosql/compare/v0.4.0...v0.5.0. The tag is backdated a bit, as to not include the most recent changes, which may have a bigger impact.

Also ran `go mod tidy`, `go mod vendor`.

### CODE REVIEW

**Comment by hslatman** [comment]
The `nosql` dependency may have been imported to a specific commit at some point, and maybe that was reverted. Or it was never updated at all. Someone on Discord had issues similar to https://github.com/smallstep/certificates/issues/743.

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   X.509 certificates.
 - Added provisioner webhooks for augmenting template data and authorizing certificate requests before signing.
 
+### Fixed
+- MySQL DSN parsing issues fixed with upgrade to [smallstep/nosql@v0.5.0](https://github.com/smallstep/nosql/releases/tag/v0.5.0).
```
Comment (dopey) [inline CHANGELOG.md:26]: should this be DNS?

**Review by dopey — COMMENTED** [review]
not sure if that's a spelling mistake

File: CHANGELOG.md (hunk 1 above)
Comment (hslatman) [reply CHANGELOG.md:26]: In`nosql` the Data Source Name (DSN) parsing/usage was fixed, so no 😅

**Review by dopey — APPROVED** [review]
lgtm

**Pull request merged**